### ISSUE TITLE
Naming temporary layers in processing algorithm dialogs

### ISSUE THREAD

**Issue by Llaves** [qgis/QGIS#50169]
### Feature description

When I run a processing algorithm that produces an output layer (eg, Dissolve) if I select output as a temporary layer, I cannot provide a name for the output layer - it will default to an algorithm dependent name, typically the same name as the algorithm. QGIS would rock even more if it provided the user the option to name the temporary output layer. For example, if I'm dissolving the county layer, I might want to name the output "Dissolved Counties" rather than just the default "Dissolve".  Of course, it's pretty easy to go into the legend pane and change the layer name, so this is not a show-stopper.

**Comment by MieWinstrup** [comment]
I think this is the same as https://github.com/qgis/QGIS/issues/29716 ?

**Comment by saberraz** [comment]
This should be addressed in https://github.com/qgis/QGIS/pull/60984 for QGIS 3.44.

**Comment by MieWinstrup** [comment]
Yes, when running a tool as a batch it works now (that's great!). 

But when entering a name in the processing, like here:

![Image](https://github.com/user-attachments/assets/591227aa-b2c7-4480-8b0a-0542fbacac78)

I get this error:
![Image](https://github.com/user-attachments/assets/33459f60-64fc-40ed-b079-fc1f671b43c3)

(This is QGIS 3.44.0)

**Comment by kewgis** [comment]
I think there's some confusion/conflation here between the ability to provide a name for a temporary output of a **single process** (not currently implemented) vs. the ability to provide a name for temporary outputs of a **batch process** (implemented by https://github.com/qgis/QGIS/pull/60984).

This FR is for the ability to provide a name for a temporary output of a **single process**.
